src/main/java/org/opendatakit/aggregate/parser/BaseFormParserForJavaRosa.java

```diff
@@ -222,53 +220,13 @@ public static String xmlWithTimestampComment(String xmlWithoutTimestampComment,
         + xmlWithoutTimestampComment.substring(idx);
   }
 
-  private static class XFormParserWithBindEnhancements extends XFormParser {
```

Comment:
This change makes me a little nervous and the commit message doesn't explain why this change was made. 

I think this code pulls the `odk:length` attribute from a binding and uses it to set the max length of the DB column. Some of this is explained here: https://opendatakit.org/help/form-design/guidelines/#datastore_string_length

Reply:
Are you sure JR has support for this attribute now? And can you confirm that forms with length work the same with this change?

Reply:
@kkrawczyk123 can you please add a test case for this in release testing? `odk:length` is really important because it controls the size of the DB column. You can read more at that link above and let me know if you want me to build you the test form.

Reply:
Of course!

Reply:
I think @yanokwa is right and this change could break this feature. I removed the class because I remembered doing safely the same on Briefcase, but it is after reading it, it's clear that Aggregate depends on reading that attribute to correctly size the storage.

The root reason to remove that class is that it depends on some method in JR that used to be accessible but now it's private. I think I can provide an alternative that doesn't rely on extension to achieve the same result.

Reply:
Maybe the issue is with MySQL? My guess is that length is used to configure `VARCHAR(n)` in the generation of tables. And performance does change based on this (see https://stackoverflow.com/a/8295254). Would we provide an method in JR for this?

Reply:
From the `FormDef` it's possible to navigate to the appropriate `TreeElement` and `getBindAttribute("odk", "length")` to read the length bind attribute.

Reply:
The other thing to look into is whether saving raw bindings might be necessary for checks done when a new version of a form is uploaded.